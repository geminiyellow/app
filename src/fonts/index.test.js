import { setupFonts, addFontForWeb, FONTS } from './'
import setupMethods from './setup'

jest.mock('./setup', () => require('method-mocks').setupMethodMocks({}), { virtual: true })

describe('.setupFonts()', () => {
  let spy

  beforeEach(() => {
    spy = setupMethods.setMethodMock('initialSetup', jest.fn())
  })

  afterEach(() => {
    setupMethods.clearAllMethodMocks()
  })

  it('calls initialSetup()', () => {
    setupFonts()

    expect(spy).toHaveBeenCalled()
  })
})

describe('.addFontForWeb()', () => {
  let spy

  beforeEach(() => {
    spy = setupMethods.setMethodMock('addWebFont', jest.fn())
  })

  afterEach(() => {
    setupMethods.clearAllMethodMocks()
  })

  it('calls addWebFont()', () => {
    addFontForWeb(1, 2, 3)

    expect(spy).toHaveBeenCalledWith(1, 2, 3)
  })
})

describe('.FONTS', () => {
  it('is correct', () => {
    expect(FONTS).toMatchSnapshot()
  })
})
