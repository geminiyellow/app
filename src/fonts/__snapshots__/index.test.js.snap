// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`.FONTS is correct 1`] = `
Object {
  "OpenSans": Object {
    "styles": Object {
      "Italic": "italic",
    },
    "weights": Object {
      "Bold": "700",
      "ExtraBold": "800",
      "Light": "300",
      "Normal": "400",
      "SemiBold": "600",
    },
  },
}
`;
